onnxruntime/core/providers/mkldnn/nn/batch_norm.cc

```diff
@@ -0,0 +1,287 @@
+// Copyright(C) 2018 Intel Corporation
+// Licensed under the MIT License
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/nn/batch_norm.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+#include "core/providers/mkldnn/memcpy_s.h"
+#include "core/providers/cpu/nn/batch_norm_helper.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN BatchNorm primitive.
+struct BatchNormParams {
+  mkldnn::memory::dims& src_dims;
+  mkldnn::memory::dims& scale_dims;
+  mkldnn::memory::dims& b_dims;
+  mkldnn::memory::dims& mean_dims;
+  mkldnn::memory::dims& var_dims;
+  mkldnn::memory::dims& dst_dims;
+  float epsilon;
+  int num_dimensions;
+
+  BatchNormParams(mkldnn::memory::dims& src_dims_mkl,
+    mkldnn::memory::dims& scale_dims_mkl,
+    mkldnn::memory::dims& b_dims_mkl, mkldnn::memory::dims& mean_dims_mkl,
+    mkldnn::memory::dims& var_dims_mkl, mkldnn::memory::dims& dst_dims_mkl,
```

Comment:
Differentiate between const-ref and just ref for those that will be modified.

Reply:
Okay. Will pass src_dims by const ref and dst_dims by ref.